erpnext/stock/doctype/packed_item/packed_item.py

```diff
@@ -63,38 +69,65 @@ def update_packing_list_item(doc, packing_item_code, qty, main_item_row, descrip
 	bin = get_bin_qty(packing_item_code, pi.warehouse)
 	pi.actual_qty = flt(bin.get("actual_qty"))
 	pi.projected_qty = flt(bin.get("projected_qty"))
+	packed_items_list.append(pi.idx)
 
-def make_packing_list(doc):
+def make_packing_list(doc, doctype=None, name=None, fieldname=None):
 	"""make packing list for Product Bundle item"""
 
 	if doc.get("_action") and doc._action == "update_after_submit": return
 
+	maintain_packed_items_list = 0
+	if doctype is not None and doctype == "Sales Order":
+		maintain_packed_items_list = cint(frappe.get_value(doctype, name, fieldname))
```

Comment:
Based on the variable name, seems like you're checking for a very specific value here.

Why not just assume its consistent here, rather than have to maintain this code in so many different places (wherever you're calling `update_packing_list_suggested_from_sales_order_invoice`)?

Reply:
Because this function is shared between sales invoice and sales order.

Reply:
Yeah, that's true. But my point is remove the `fieldname` argument and directly use `maintain_packed_items_list`, since it's too specific to ever change.

More code to maintain means more scope for error.

Reply:
Ok I got it